### ISSUE TITLE
RSocket support for Unity3D

### ISSUE THREAD

**Issue by ghost** [rsocket/rsocket-net#32]
As for now, it seems not to be possible to use rsocket-net from inside Unity3D project.
RSocket seems like an ideal solution for some kind of multiplayer games (turn-based, some casual networking games). Game DEV industry might be an area for potential growth and adoption.

## Motivation
I created a multiplayer, turned-based game for Android and so I tried to use RSocket as it seemed the most appropriate solution for my needs. I tried gRPC, Websockets, SocketIO etc. and ended up with SocketIO, but I still regret not investing time for making RSocket.NET run on my project somehow.

## Desired solution
It would be great if the library would be supported for all platform like WebGL, Android, IOS, consoles, PC.
I know RSocket is transport-transparent by design and it would make potential rsocket-unity3d project easier to implement (ie. WebGL supports only Websockets). The more significant problem (potentially) would be to make it Android compatible (AOT issues).
So, in a nutshell, the desired solution would be to create a separate project that would use only Unity3D compatible libraries and would support both Android and WebGL (as AFAIK only these platforms cause problems).

## Considered alternatives
I am not sure, but it may be possible to modify rsocket-net project so that it could be ran on Unity3D.
